.github/workflows/build-and-test.yml

```diff
@@ -8,92 +8,166 @@ on:
         required: true
         type: string
 
-env:
-  DOCKER_BUILDER_IMAGE: mangatasolutions/node-builder:0.2
-  DISABLE_TTY: 1
+permissions:
+  contents: write
+  id-token: write
 
 jobs:
   build-node-image:
+    name: Build Docker image
     runs-on: [self-hosted, compile]
+    container:
+      image: mangatasolutions/node-builder:multi
     steps:
       - uses: actions/checkout@v3
-      - uses: docker/setup-buildx-action@v2.2.1
-      - uses: docker/login-action@v2.1.0
+      - name: Authenticate to Google Cloud
+        uses: google-github-actions/auth@v1
         with:
-          username: ${{ secrets.DOCKERHUB_USERNAME }}
-          password: ${{ secrets.DOCKERHUB_TOKEN }}
+          workload_identity_provider: '${{ secrets.GCP_WORKLOAD_IDENTITY_PROVIDER }}'
+          service_account: '${{ secrets.GCP_SERVICE_ACCOUNT }}'
 
-      - name: Build and push
-        uses: docker/build-push-action@v3.2.0
+      - name: Cache the Cargo dependencies
+        uses: mansagroup/gcs-cache-action@v1.0.3
+        with:
+          bucket: mangata-node-ci-cache
+          path: |
+            ${{ github.workspace }}/target
+            /usr/local/cargo/git
+            /usr/local/cargo/registry
+            ~/.cache/sccache
+          key: compile-cache-${{ runner.os }}-${{ hashFiles('Cargo.lock') }}
+      
+      - name: Compile mangata-node code
+        run: cargo build --release --no-default-features --features=mangata-rococo,mangata-kusama,runtime-benchmarks
```

Comment:
is that executed inside the docker or using native compiler installed on gcp hosts ?

Reply:
>It is executed by the container that runs from this image:
cool!

>can you confirm we are fine with using this toolchain option?
yes, thats fine, but i cant see `toolchain` option arg in above command. Also keep in mind that used toolchain should be relatively easy to change as with parity is bumping it version with almost every substrate framework release